test/auth_test.py

```diff
@@ -209,52 +213,73 @@ def test_transfered_ongoing_non_continues_ops(cfg):
 
 
 @pytest.mark.benchmark
-def test_run_operation_benchmark(cfg):
-    # Run 1000000 operations with 4 concurrent threads.
+@pytest.mark.parametrize("workers", [1, 2, 4, 8])
+def test_run_operation_benchmark(cfg, workers):
+    # Simulate tranferr of a big image using nbdcopy as a client. We split the
+    # image to 128 MiB segments. Every wroker send one segment and get the next
+    # segment to transfer.
+
     ticket = Ticket(testutil.create_ticket(ops=["read"]), cfg)
-    operations = 10**6
-    workers = 4
-    chunk = 10**9
-    step = chunk * workers // operations
 
-    def worker(offset, size):
-        while offset < size:
-            ticket.run(Operation(offset, step))
-            offset += step
+    image_size = 100 * 1024**3
+    segment_size = 128 * 1024**2
+    io_size = 2 * 1024**2
+
+    segment_count = (image_size + segment_size - 1) / segment_size
+    workers = min(workers, segment_count)
+
+    counter = itertools.count()
+
+    def next_segment():
+        n = next(counter)
+        if n < segment_count:
+            return n
+
+    def worker():
+        for segment in iter(next_segment, None):
+            offset = segment * segment_size
+            end = min(offset + segment_size, image_size)
+
+            log.info("Transferring segment %d offset %d", segment, offset)
+            while offset < end:
+                n = min(io_size, end - offset)
+                ticket.run(Operation(offset, n))
+                offset += n
 
     start = time.monotonic()
 
     threads = []
     try:
         for i in range(workers):
-            t = util.start_thread(worker, args=(i * chunk, chunk))
+            t = util.start_thread(worker, name=f"run/{i}")
             threads.append(t)
     finally:
         for t in threads:
             t.join()
 
     elapsed = time.monotonic() - start
+    iops = image_size / io_size
```

Comment:
Again, wouldn't you like to have an integer here, so need to use "//"?

Reply:
Yes, this is a bug, should be //. It works because I chose size which
is multiple of io_size.